### PR TITLE
Feature/move platform version to cdap api

### PR DESCRIPTION
**Pull request by sechegaray**
First step of moving ProjectVersion to cdap-api.
Copying ProjectVersion to cdap-api, and renaming it PlatformVersion

### CODE REVIEW

File: cdap-api/pom.xml [hunk 1]
```diff
@@ -29,6 +29,11 @@
   <artifactId>cdap-api</artifactId>
   <name>CDAP API</name>
   <packaging>jar</packaging>
+  <properties>
+    <!-- Workaround for bug http://jira.codehaus.org/browse/MRESOURCES-99 -->
```
Comment (chtyim) [inline cdap-api/pom.xml]: First, the link seems not working. It should be https://issues.apache.org/jira/browse/MRESOURCES-99 ?

Also, can we just use the latest maven resources plugin (3.2.0)?
Third, why we need timestamp? The existing code is using `project.info.buildtime`?

File: cdap-api/pom.xml [hunk 2]
```diff
@@ -66,4 +71,38 @@
...
+        <filtering>false</filtering>
+        <includes>
+          <include>**/*.xml</include>
+        </includes>
+      </resource>
+    </resources>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-jar-plugin</artifactId>
```
Comment (chtyim) [inline cdap-api/pom.xml]: Why this is needed?

File: cdap-api/pom.xml (hunk 1 above)
Comment (sechegaray) [reply cdap-api/pom.xml]: Updated the URL.

I tested using Maven Resources plugin (3.2.0). While I was able to expand maven.build.timestamp in the properties file. I was not able to change its formatting to include milliseconds.

File: cdap-api/pom.xml (hunk 2 above)
Comment (sechegaray) [reply cdap-api/pom.xml]: Removed.

File: cdap-api/pom.xml [hunk 3]
```diff
@@ -66,4 +73,29 @@
     </dependency>
   </dependencies>
 
+  <build>
+    <!-- Resource filtering for non xml files only -->
+    <resources>
+      <resource>
+        <directory>src/main/resources</directory>
+        <filtering>true</filtering>
+        <excludes>
```
Comment (chtyim) [inline cdap-api/pom.xml]: We don't have XML in cdap-api. You can remove this excludes block.

File: cdap-api/pom.xml [hunk 4]
```diff
@@ -66,4 +73,29 @@
...
+          <exclude>**/*.xml</exclude>
+        </excludes>
+      </resource>
+      <resource>
```
Comment (chtyim) [inline cdap-api/pom.xml]: We don't have XML in cdap-api. You can remove this block.

File: cdap-api/pom.xml [hunk 5]
```diff
@@ -66,4 +73,29 @@
...
+        <directory>src/main/resources</directory>
+        <filtering>false</filtering>
+        <includes>
+          <include>**/*.xml</include>
+        </includes>
+      </resource>
+    </resources>
+    <plugins>
```
Comment (chtyim) [inline cdap-api/pom.xml]: I think you don't need this block.

**Review by chtyim — APPROVED** [review]
Just some comments around cleaning up the pom.xml file.

**Pull request merged**